web/server/web_client.c

```diff
@@ -1906,6 +1925,15 @@ ssize_t web_client_receive(struct web_client *w)
     if ( (!web_client_check_unix(w)) && (netdata_ssl_srv_ctx) ) {
         if ( ( w->ssl.conn ) && (!w->ssl.flags)) {
             bytes = netdata_ssl_read(w->ssl.conn, &w->response.data->buffer[w->response.data->len], (size_t) (left - 1));
+            int ssl_err = SSL_get_error(w->ssl.conn, bytes);
```

Comment:
The new algorithm present here is also repeated two times above, I suggest to create a function, probably an inline function to avoid this repetition.

Reply:
Good suggestion, thanks Thiago, done!